### PR TITLE
Fix annotations not shown bug for submissions outside a course

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request fixes a bug where annotations were not shown in the code tab for submissions with course_id nil.

### CODE REVIEW

File: app/helpers/renderers/feedback_code_renderer.rb [hunk 1]
```diff
@@ -92,7 +92,7 @@ def add_messages(submission, messages, user)
     @builder.script(type: 'application/javascript') do
       @builder << <<~HEREDOC
         window.MathJax.startup.promise.then(() => {
-          window.dodona.codeListing = new window.dodona.codeListingClass(#{submission.id}, #{submission.course_id}, #{submission.exercise_id}, #{user.id}, #{@code.to_json}, #{@code.lines.length}, #{user_is_student});
+          window.dodona.codeListing = new window.dodona.codeListingClass(#{submission.id}, #{submission.course_id || 'undefined'}, #{submission.exercise_id}, #{user.id}, #{@code.to_json}, #{@code.lines.length}, #{user_is_student});
```
Comment (chvp) [inline app/helpers/renderers/feedback_code_renderer.rb:95]: Maybe use `submission.course_id.to_json`?

**Review by chvp — APPROVED** [review]

File: app/helpers/renderers/feedback_code_renderer.rb [hunk 2]
```diff
@@ -92,7 +92,7 @@ def add_messages(submission, messages, user)
     @builder.script(type: 'application/javascript') do
       @builder << <<~HEREDOC
         window.MathJax.startup.promise.then(() => {
-          window.dodona.codeListing = new window.dodona.codeListingClass(#{submission.id}, #{submission.course_id}, #{submission.exercise_id}, #{user.id}, #{@code.to_json}, #{@code.lines.length}, #{user_is_student});
+          window.dodona.codeListing = new window.dodona.codeListingClass(#{submission.id}, #{submission.course_id.to_json}, #{submission.exercise_id}, #{user.id}, #{@code.to_json}, #{@code.lines.length}, #{user_is_student});
```
Comment (niknetniko) [inline app/helpers/renderers/feedback_code_renderer.rb:95]: We should probably also update the types in `code_listing.ts` that the course id can now be `null`  instead of `undefined`  (although looking at that code, the types don't indicate it can be `undefined` at all, so perhaps this should be done in another PR).
